markdown/issues.md

```diff
@@ -1,10 +1,82 @@
+We use Caddy as a web server as it is used in Tutor both as a web proxy and for generating SSL/TLS certificates at runtime.
+
+We were looking for a solution to expose the caddy service other than the loadbalancer and make our platform Octavia free. According to the Kubernetes documentation, several ways exist to expose a service in Kubernetes.
+
+* The first approach, ClusterIP, allows the service to be accessed only via an internal IP within the cluster. However, this type of exposure is not suitable for a production environment as it restricts accessibility within the cluster.
+
+* The second option, NodePort, is also unsuitable for our needs since our Caddy service operates on ports 80 and 443, while the NodePort range is from 30000 to 32767.
+
+* The third option is a loadbalancer, which we are trying to eliminate.
+* Lastly, there is a way not widely used among the kubernetes community because many are using cloud provider's Load balancers.
+Using ClusterIP with external IP
+
+Customers can expose their Kubernetes cluster to external traffic using ClusterIP and an external IP address. The Kubernetes service type is ClusterIP, but an external IP address can be specified using the externalIP property. This solution requires that the customer maps the external IP address of a service to their private IP addresses in the cluster. 
+
+Unfortunately, we encountered issues in utilizing a Floating IP address as an external IP address. It was due to the fact that our Floating IP address was mapped to a private address within a different subnet than the one used by the ClusterIP.
+
+So, we were not able to figure out any way to expose our services without Loadbalancer.
 
 
 ## Orphaned Pods
+
+<!-- Note -->
+
+What are Orphaned pods in a Kubernetes cluster?
+
+Orphaned pods can occur in Kubernetes when their owner objects, such as a deployment, replica set, or replication controller, is deleted or modified.
+By default, Kubernetes deletes these dependent objects. Therefore, the responsibility of cleaning up these objects lies with the kubelet.
+
+What is kubelet?
+
+The kubelet is the primary "node agent" that runs on each node.
+Kubelet is responsible for applying, creating, updating, and destroying containers on a Kubernetes node. It performs garbage collection on unused images and unused containers on the nodes.
+
+When kubelet fails to delete these dependent objects, orphaned pods remain dangling in the environment and interfering with the deployment process.
+
+We are facing this issue while performing a rolling update in our deployment.
+
+What happens in a rolling deployment?
+During a rolling deployment, the controller deletes and recreates pods with the updated configuration, one by one, without causing downtime to the cluster.
+
+However, our rolling deployment gets stuck when the pod running the old version remains in the Terminating state, and the new version that needs to be recreated gets stuck in the ContainerCreating state.
```

Comment:
You must add here that this is only a problem because some of the affected Pods mount volumes that are `RWO`, rather than `RWX`. If we could use `RWX` volumes, this wouldn't be a problem — and it *isn't* more than a cosmetic problem for Pods not using any volumes at all (of which we also have several)